Sources/FoundationEssentials/String/String+Path.swift

```diff
@@ -512,28 +468,81 @@ extension String {
+        // If all else fails, fall back to /var/empty
         return "/var/empty"
-        #endif // os(Windows)
+        #endif
+    }
+
+    internal static func homeDirectoryPath(forUser user: String) -> String? {
+        #if os(Windows)
+        guard !user.isEmpty else {
+            return nil
+        }
+        
+        return user.withCString(encodedAs: UTF16.self) { pwszUserName in
+            var cbSID: DWORD = 0
+            var cchReferencedDomainName: DWORD = 0
+            var eUse: SID_NAME_USE = SidTypeUnknown
+            LookupAccountNameW(nil, pwszUserName, nil, &cbSID, nil, &cchReferencedDomainName, &eUse)
+            guard cbSID > 0 else {
+                return nil
+            }
+            
+            return withUnsafeTemporaryAllocation(of: CChar.self, capacity: Int(cbSID)) { pSID in
+                return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(cchReferencedDomainName)) { pwszReferencedDomainName in
+                    guard LookupAccountNameW(nil, pwszUserName, pSID.baseAddress, &cbSID, pwszReferencedDomainName.baseAddress, &cchReferencedDomainName, &eUse) else {
+                        return nil
+                    }
+                    
+                    var pwszSID: LPWSTR? = nil
+                    guard ConvertSidToStringSidW(pSID.baseAddress, &pwszSID) else {
+                        fatalError("unable to convert SID to string for user \(user)")
+                    }
+                    
+                    return #"SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\\#(String(decodingCString: pwszSID!, as: UTF16.self))"#.withCString(encodedAs: UTF16.self) { pwszKeyPath in
+                        return "ProfileImagePath".withCString(encodedAs: UTF16.self) { pwszKey in
+                            var cbData: DWORD = 0
+                            RegGetValueW(HKEY_LOCAL_MACHINE, pwszKeyPath, pwszKey, RRF_RT_REG_SZ, nil, nil, &cbData)
+                            guard cbData > 0 else {
+                                fatalError("unable to query ProfileImagePath for user \(user)")
+                            }
+                            return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(cbData)) { pwszData in
+                                guard RegGetValueW(HKEY_LOCAL_MACHINE, pwszKeyPath, pwszKey, RRF_RT_REG_SZ, nil, pwszData.baseAddress, &cbData) == ERROR_SUCCESS else {
+                                    fatalError("unable to query ProfileImagePath for user \(user)")
+                                }
+                                return String(decodingCString: pwszData.baseAddress!, as: UTF16.self)
+                            }
+                        }
+                    }
+                    
+                }
+            }
+        }
+        #else
+        // First check CFFIXED_USER_HOME if the environment is not considered tainted
+        if let envVar = Platform.getEnvSecure("CFFIXED_USER_HOME") {
```

Comment:
Just noticed this. The behavior appears consistent with `CFCopyHomeDirectoryURLForUser`, but I'm not sure it's correct. 😓 An issue for a different day, perhaps.

Reply:
Yeah it did seem a bit weird that `CFFIXED_USER_HOME` affects all users and not just the current user, but I agree probably an issue to address separately to minimize the behavioral changes of this patch